### PR TITLE
additional panel meta (is_new, beta, category)

### PR DESCRIPTION
**Pull request by ritch**
Here's an example using all of the new settings in python and JS:

```python
class CounterExample(foo.Panel):
    @property
    def config(self):
        return foo.PanelConfig(
            name="example_counter",
            label="Examples: Counter",
            category=foo.Categories.ANALYZE, # new
            is_new=foo.is_new("2024-11-01"), # new
            beta=True # new,
            priority=52000 # new
        )
```

```typescript
registerComponent({
  name: "Embeddings",
  label: "Embeddings",
  component: Embeddings,
  type: PluginComponentType.Panel,
  activator: () => true,
  Icon: WorkspacesIcon,
  panelOptions: {
    TabIndicator: EmbeddingsTabIndicator,
    priority: BUILT_IN_PANEL_PRIORITY_CONST,
    category: Categories.Curate, // new
    isNew: true, // new
    beta: true, // new
    priority: 51001 // existing
  },
});
```

New information is displayed only in the `AddPanel*` component.

image
image

## Release Notes

- **New Features**
	- Introduced a categorization system for panels, allowing users to easily identify and organize panels by their respective categories.
	- Added visual indicators for "NEW" and "BETA" features in the UI, enhancing user awareness of recent additions and experimental features.

- **Improvements**
	- Enhanced the `AddPanelButton` and `AddPanelItem` components for better usability and organization.
	- Improved panel registration capabilities with new metadata options, including priority and categorization.

- **Bug Fixes**
	- Minor adjustments to ensure consistent behavior across components.

### CODE REVIEW

**Review by minhtuev — COMMENTED** [review]
Thanks Ritchie, meta info look good for me, not sure how to determine priority value thoug.

File: app/packages/plugins/src/index.ts [hunk 1]
```diff
@@ -297,6 +297,37 @@ export enum PluginComponentType {
    */
 }
 
+type CategoryID = "import" | "curate" | "analyze" | "custom";
```
Comment (Br2850) [inline app/packages/plugins/src/index.ts:300]: Given that @brimoor this was your original vision for the categories of workflows with Assistant, do we want to simply just add all of these at right now?

Not sure if this list is still the "right" list in terms of categorization, but food for thought @ritch 

Screenshot 2024-11-04 at 3 13 35 PM

File: fiftyone/operators/utils.py [hunk 2]
```diff
@@ -66,3 +67,40 @@ def is_method_overridden(base_class, sub_class_instance, method_name):
...
+            Defaults to 30.
+
+    Returns:
+        bool: True if the release date is within the specified number of days, False otherwise.
+
+    Examples:
+        >>> is_new("2024-11-09")  # Using a string date and default days
+        True  # if today's date is within 30 days after 2024-11-09
+
+        >>> is_new(datetime(2024, 11, 9), days=15)  # Using a datetime object with 15-day threshold
```
Comment (Br2850) [inline fiftyone/operators/utils.py:95]: That's cool, I like that we can manually define when `NEW` should disappear 🙌🏽

**Review by Br2850 — APPROVED** [review]
Looks great! 🎉 Similar sentiment to Minh' not sure how we would self define priority - maybe it should be a calculation of (new + category, etc.) but other than that this is super awesome!

File: app/packages/plugins/src/index.ts (hunk 1 above)
Comment (ritch) [reply app/packages/plugins/src/index.ts:300]: We've been discussing the list in a slack thread

**Pull request merged**